src/app/clusters/laundry-washer-controls-server/laundry-washer-controls-server.cpp

```diff
@@ -0,0 +1,191 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/config.h>
+
+#include "laundry-washer-controls-delegate.h"
+#include "laundry-washer-controls-server.h"
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/callback.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/enums.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/server/Server.h>
+#include <app/util/error-mapping.h>
+#include <lib/core/CHIPEncoding.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::LaundryWasherControls;
+using namespace chip::app::Clusters::LaundryWasherControls::Attributes;
+using chip::Protocols::InteractionModel::Status;
+
+static constexpr size_t kLaundryWasherControlsDelegateTableSize =
+    EMBER_AF_LAUNDRY_WASHER_CONTROLS_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+// -----------------------------------------------------------------------------
+// Delegate Implementation
+//
+namespace {
+Delegate * gDelegateTable[kLaundryWasherControlsDelegateTableSize] = { nullptr };
+}
+
+namespace {
+Delegate * GetDelegate(EndpointId endpoint)
+{
+    return (endpoint > kLaundryWasherControlsDelegateTableSize ? nullptr : gDelegateTable[endpoint]);
+}
+
+} // namespace
+
+LaundryWasherControlsServer LaundryWasherControlsServer::sInstance;
+
+/**********************************************************
+ * LaundryWasherControlsServer public methods
+ *********************************************************/
+void LaundryWasherControlsServer::SetDefaultDelegate(EndpointId endpoint, Delegate * delegate)
+{
+    if (endpoint < kLaundryWasherControlsDelegateTableSize)
```

Comment:
Again, the index into the list is _not_ the endpoint id.  Please see other clusters that have such delegate lists.

Reply:
Will change it